src/v/cloud_topics/level_one/frontend_reader/reader.cc

```diff
@@ -0,0 +1,439 @@
+/*
+ * Copyright 2025 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+#include "cloud_topics/level_one/frontend_reader/reader.h"
+
+#include "bytes/iostream.h"
+#include "cloud_topics/data_plane_api.h"
+#include "cloud_topics/logger.h"
+#include "config/configuration.h"
+#include "model/timeout_clock.h"
+
+#include <exception>
+#include <iterator>
+#include <utility>
+
+namespace cloud_topics {
+
+// TODO: Add config.
+static constexpr size_t L1_max_bytes_per_object_fetch = 1024_KiB;
+
+level_one_log_reader_impl::level_one_log_reader_impl(
+  cloud_topic_log_reader_config& cfg,
+  model::ntp ntp,
+  model::topic_id_partition tidp,
```

Comment:
why do we need both ntp and tidp here?

Reply:
I think we need to update the l1 object builder/reader to use topic ids instead of ntps. I was going to do that and forgot

Reply:
https://github.com/redpanda-data/redpanda/pull/27453